File: exercice-5/main.ts
class ConcertTicket {

  public artistName: string;
  public placeNumber: string;
  public date: string;
  public hour: number;
  public price: string = "";

  constructor(artistName: string, placeNumber: string, date: string, hour: number, price?: string) {

    this.artistName = artistName;
    this.placeNumber = placeNumber;
    this.date = date;
    this.hour = hour;
    if (price) {
      this.price = price
    }
  }


  public htmlDetail(): string {
    let htmlContent: string = `
  <p>Nom de l'artiste : ${this.artistName} </p>
  <p>Numero de place : ${this.placeNumber}</p>
  <p>Jour du concert : ${this.date}</p>
  <p> Heure du concert : ${this.hour}</p> `;
    if (this.price != "") {
      htmlContent = htmlContent + `<p> Prix du concert : ${this.price}</p> `;
    };
    return htmlContent;
  }
}

let ticket01: ConcertTicket = new ConcertTicket("Franky Vincent", "23", "15/10/2008", 21);
console.log(ticket01.htmlDetail())
let ticket02: ConcertTicket = new ConcertTicket("Helene Segara", "01", "01/02/2020", 21, "23.90");
console.log(ticket02.htmlDetail())

File: exercice-5/main.java
 
class Playground {
    public static void main(String[ ] args) {
        Ticket ticketMadonna = new Ticket(1, "Madonna", "22 Février 2019", "20h00", "Le Dome", "95€");
        Ticket ticketTheWeeknd = new Ticket(2, "The Weeknd", "15 Novembre", "21h00", "Accor Arena");
        System.out.println(ticketTheWeeknd.ticketInformations());
    }
}

public class Ticket {
    public int id;
    public String artist;
    public String date;
    public String hour;
    public String location;
    public String price = "";

    public Ticket(int id, String artist, String date, String hour, String location, String price ){
        this.id = id;
        this.artist = artist;
        this.date = date;
        this.hour = hour;
        this.location = location;
        this.price = price;
    }

    public Ticket(int id, String artist, String date, String hour, String location){
        this.id = id;
        this.artist = artist;
        this.date = date;
        this.hour = hour;
        this.location = location;
    }

    String ticketInformations(){
        String details = "<p>Nom de l'artiste " + this.artist + "</p> <p>Id du billet : " + this.id + "</p><p>date du concert : " + this.date + "</p><p> Heure du concert : " + this.hour + " </p> <p>Lieu du concert : " + this.location + "</p>";

        if (this.price != "" ) {
            details = details + "<p> </p> prix du billet : " + this.price;
        }

        return details;
        }
}